### ISSUE TITLE
Footnotes numbers are wrongly-placed in RTL HTML output

### ISSUE THREAD

**Issue by matanhakim** [jgm/pandoc#9064]
### Bug description
When rendering a markdown document which is Right-To-Left (for me, in Hebrew), the footnotes numbers (i.e., where they appear in the body of the text, and not in the footer) are wrongly-placed on the right-hand-side when they come after a number, as if it was written Left-To-Right.
This happens in HTML output.

I imagine this has something to do with the [Bidirectional Algorithm](https://www.w3.org/International/articles/inline-bidi-markup/uba-basics), but this is really out of my reach to figure this out.
Also, I guess #6222 and #6011 have something to do with it.

### Steps to reproduce
I took the following `example.md` document:

```
---
title: "RTL Issues with footnotes"
dir: rtl
---

זה טקסט בעברית שכתוב היטב מימין לשמאל.

(This is Hebrew text written well from right to left.)

כשנוספת הערת שוליים[^1], היא מוצגת בצורה טובה. גם כשהיא מגיעה בסוף משפט.[^2]

[^1]: הערה ראשונה

[^2]: הערה שנייה

(When you add a footnote, it is presented well. Even when it comes at the end of a sentence.)

לעומת זאת, אם היא מגיעה אחרי מספר, היא מוצגת בצד הלא נכון, כמו בשנת 2023.[^3]

[^3]: הערה שלישית

(Nevertheless, if it comes after a number, it is presented on the wrong side, like in the year 2023.)
```

and ran the following command line:
```
pandoc --to html -s -o example.html example.md
```

The HTML output is as follows:
![image](https://github.com/jgm/pandoc/assets/111883851/fc98d283-19be-4d1f-ba52-0b912402670b)
Please notice that footnote number 3 is on the right-hand side of the "2023.", where it shouldn't be.

This issue is a follow-up on a Quarto issue quarto-dev/quarto-cli#6739

### Environment
Windows 10 Home
Pandoc 3.1.7

Would appreciate any help!

**Comment by tarleb** [comment]
I agree with your analysis that this is due to the bidirectional algorithm applied by the browser. The workaround is to explicitly mark the year as being ltr by wrapping it in a span with the attribute `dir=ltr`.

``` markdown
לעומת זאת, אם היא מגיעה אחרי מספר, היא מוצגת בצד הלא נכון, כמו בשנת [2023]{dir=ltr}.[^3]
```

I don't think this needs fixing in pandoc, but I'm not 100% sure.
